patches/minecraft/net/minecraft/world/chunk/Chunk.java.patch

```diff
@@ -26,7 +26,32 @@
                      {
                          this.field_76634_f[k << 4 | j] = l;
  
-@@ -452,12 +453,13 @@
+@@ -274,8 +275,23 @@
+     {
+         this.field_76637_e.field_72984_F.func_76320_a("recheckGaps");
+ 
+-        if (this.field_76637_e.func_175697_a(new BlockPos(this.field_76635_g * 16 + 8, 0, this.field_76647_h * 16 + 8), 16))
++        // FORGE: require 5x5 chunks to be loaded and 3x3 chunks to be lit
++        boolean skip = !this.field_150814_l;
++
++        for (int dz = -2; dz <= 2 && !skip; dz++)
+         {
++            for (int dx = -2; dx <= 2 && !skip; dx++)
++            {
++                Chunk chunk;
++                if ((dz != 0 || dx != 0) && ((chunk = field_76637_e.func_72863_F().func_186026_b(field_76635_g + dx, field_76647_h + dz)) == null || Math.abs(dz) != 2 && Math.abs(dx) != 2 && !chunk.field_150814_l))
```

Comment:
Please move this logic into a helper function in `net.minecraftforge.common.ForgeHooks` so that it is not in the patch and can be written more simply.